File: Unit_tests_python/EX-01-compare_directories/Copilot.py
import os
import filecmp

def compare_directories(dir1, dir2):
    report = {
        'Passed': True,
        'Failed_Count': 0,
        'Failed_Tests': []
    }

    for root, dirs, files in os.walk(dir1):
        for file in files:
            file_path1 = os.path.join(root, file)
            file_path2 = os.path.join(dir2, os.path.relpath(file_path1, dir1))

            if not os.path.exists(file_path2):
                report['Passed'] = False
                report['Failed_Count'] += 1
                report['Failed_Tests'].append({
                    'Test': 'File Missing',
                    'Failed_Files': [file],
                    'Failure_Location': 'dir2'
                })
            elif not filecmp.cmp(file_path1, file_path2):
                report['Passed'] = False
                report['Failed_Count'] += 1
                report['Failed_Tests'].append({
                    'Test': 'File Content Mismatch',
                    'Failed_Files': [file],
                    'Failure_Location': 'dir1' if os.path.getmtime(file_path1) < os.path.getmtime(file_path2) else 'dir2'
                })

    return report

File: Unit_tests_lua/EX-01-compare_directories/Copilot.lua
function compare_directories(dir1, dir2)
    local report = {
        Passed = true,
        Failed_Count = 0,
        Failed_Tests = {}
    }

    -- Get the list of files in dir1
    local files1 = {}
    for file in io.popen('dir /b "'..dir1..'"'):lines() do
        table.insert(files1, file)
    end

    -- Get the list of files in dir2
    local files2 = {}
    for file in io.popen('dir /b "'..dir2..'"'):lines() do
        table.insert(files2, file)
    end

    -- Compare the files in dir1 and dir2
    for _, file in ipairs(files1) do
        if not table.contains(files2, file) then
            report.Passed = false
            report.Failed_Count = report.Failed_Count + 1
            table.insert(report.Failed_Tests, {
                Failed_Files = {file},
                Failure_Location = 'dir1'
            })
        end
    end

    for _, file in ipairs(files2) do
        if not table.contains(files1, file) then
            report.Passed = false
            report.Failed_Count = report.Failed_Count + 1
            table.insert(report.Failed_Tests, {
                Failed_Files = {file},
                Failure_Location = 'dir2'
            })
        end
    end

    return report
end
